Bug in uart_renesas_ra_sau.c ?

### Describe the bug

I observed compilation error while trying to build  samples/basic/blinky for Renesas' board fpb_ra0e1.

I fixed it by turning the function in file drivers/serial/uart_renesas_ra_sau.c line 311
from :
```
static int uart_renesas_ra_sau_irq_update(const struct device *dev)
{
	ARG_UNUSED(dev);

	return 1;
}
```

to :
```
static void uart_renesas_ra_sau_irq_update(const struct device *dev)
{
	ARG_UNUSED(dev);
}
```

### Steps to reproduce

west build -p always -b fpb_ra0e1 samples/basic/blinky

### Relevant log output

```shell
zephyrproject/zephyr/drivers/serial/uart_renesas_ra_sau.c:421:23: error: initialization of 'void (*)(const struct device *)' from incompatible pointer type 'int (*)(const struct device *)' [-Wincompatible-pointer-types]
  421 |         .irq_update = uart_renesas_ra_sau_irq_update,
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/zephyrproject/zephyr/drivers/serial/uart_renesas_ra_sau.c:421:23: note: (near initialization for 'uart_renesas_ra_sau_driver_api.irq_update')
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS : Linux
- toolchain : Zephyr SDK
- commit related : dc4e0928302

### Additional Context

_No response_

--- a/drivers/serial/uart_renesas_ra_sau.c
+++ b/drivers/serial/uart_renesas_ra_sau.c
@@ -308,13 +308,6 @@ static void uart_renesas_ra_sau_irq_err_disable(const struct device *dev)
 	cfg->regs->SCR[cfg->rx_channel] &= ~R_SAU0_SCR_EOC_Msk;
 }
 
-static int uart_renesas_ra_sau_irq_update(const struct device *dev)
-{
-	ARG_UNUSED(dev);
-
-	return 1;
-}
-
 static int uart_renesas_ra_sau_irq_is_pending(const struct device *dev)
 {
 	const struct uart_renesas_ra_sau_config *cfg = dev->config;
@@ -418,7 +411,6 @@ static DEVICE_API(uart, uart_renesas_ra_sau_driver_api) = {
 	.irq_err_enable = uart_renesas_ra_sau_irq_err_enable,
 	.irq_err_disable = uart_renesas_ra_sau_irq_err_disable,
 	.irq_is_pending = uart_renesas_ra_sau_irq_is_pending,
-	.irq_update = uart_renesas_ra_sau_irq_update,
 	.irq_callback_set = uart_renesas_ra_sau_irq_callback_set,
 #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 };

